From 821a24b12631374d33e3054783edbed7abe728f6 Mon Sep 17 00:00:00 2001
From: Vipul Kumar <vipul_kumar@mentor.com>
Date: Mon, 18 Mar 2019 17:09:42 +0530
Subject: [PATCH 2/2] Revert "MLK-12623-03 ARM: imx: Add cpu speed grading
 check for imx6ul"

commit  4eb59d329a690f5f63209caeed7dd08977740894 from
https://source.codeaurora.org/external/imx/linux-imx.git

This reverts commit 301ac2d01008323ff30f7c5e24dc40ba699b3cad.
As per 'commit 5028f5d2b38e ("cpufreq: imx6q: add 696MHz operating point
for i.mx6ul")', cpu speed grading check for imx6ul is already upstream,
so removing duplicate code.

Signed-off-by: Vipul Kumar <vipul_kumar@mentor.com>
Signed-off-by: Xiaolei Wang <Xiaolei.Wang@windriver.com>
---
 arch/arm/mach-imx/mach-imx6ul.c | 97 +--------------------------------
 1 file changed, 2 insertions(+), 95 deletions(-)

diff --git a/arch/arm/mach-imx/mach-imx6ul.c b/arch/arm/mach-imx/mach-imx6ul.c
index 99eb8ad811cb..d1a50b1c2131 100644
--- a/arch/arm/mach-imx/mach-imx6ul.c
+++ b/arch/arm/mach-imx/mach-imx6ul.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2015-2016 Freescale Semiconductor, Inc.
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -9,10 +9,8 @@
 #include <linux/mfd/syscon.h>
 #include <linux/mfd/syscon/imx6q-iomuxc-gpr.h>
 #include <linux/micrel_phy.h>
-#include <linux/of_address.h>
 #include <linux/of_platform.h>
 #include <linux/phy.h>
-#include <linux/pm_opp.h>
 #include <linux/regmap.h>
 #include <asm/mach/arch.h>
 #include <asm/mach/map.h>
@@ -68,95 +66,6 @@ static void __init imx6ul_enet_phy_init(void)
 				   0xffffffff, ksz8081_phy_fixup);
 	}
 }
-
-#define OCOTP_CFG3			0x440
-#define OCOTP_CFG3_SPEED_SHIFT		16
-#define OCOTP_CFG3_SPEED_696MHZ		0x2
-#define OCOTP_CFG3_SPEED_900MHZ		0x3
-
-static void __init imx6ul_opp_check_speed_grading(struct device *cpu_dev)
-{
-	struct device_node *np;
-	void __iomem *base;
-	u32 val;
-
-	if (cpu_is_imx6ul())
-		np = of_find_compatible_node(NULL, NULL, "fsl,imx6ul-ocotp");
-	else
-		np = of_find_compatible_node(NULL, NULL, "fsl,imx6ull-ocotp");
-
-	if (!np) {
-		pr_warn("failed to find ocotp node\n");
-		return;
-	}
-
-	base = of_iomap(np, 0);
-	if (!base) {
-		pr_warn("failed to map ocotp\n");
-		goto put_node;
-	}
-
-	/*
-	 * Speed GRADING[1:0] defines the max speed of ARM:
-	 * 2b'00: Reserved;
-	 * 2b'01: 528000000Hz;
-	 * 2b'10: 700000000Hz(i.MX6UL), 800000000Hz(i.MX6ULL);
-	 * 2b'11: 900000000Hz(i.MX6ULL);
-	 * We need to set the max speed of ARM according to fuse map.
-	 */
-	val = readl_relaxed(base + OCOTP_CFG3);
-	val >>= OCOTP_CFG3_SPEED_SHIFT;
-	val &= 0x3;
-	if (cpu_is_imx6ul()) {
-		if (val < OCOTP_CFG3_SPEED_696MHZ) {
-			if (dev_pm_opp_disable(cpu_dev, 696000000))
-				pr_warn("Failed to disable 696MHz OPP\n");
-		}
-	}
-
-	if (cpu_is_imx6ull()) {
-		if (val != OCOTP_CFG3_SPEED_696MHZ) {
-			if (dev_pm_opp_disable(cpu_dev, 792000000))
-				pr_warn("Failed to disable 792MHz OPP\n");
-		}
-
-		if (val != OCOTP_CFG3_SPEED_900MHZ) {
-			if(dev_pm_opp_disable(cpu_dev, 900000000))
-				pr_warn("Failed to disable 900MHz OPP\n");
-		}
-	}
-	iounmap(base);
-
-put_node:
-	of_node_put(np);
-}
-
-static void __init imx6ul_opp_init(void)
-{
-	struct device_node *np;
-	struct device *cpu_dev = get_cpu_device(0);
-
-	if (!cpu_dev) {
-		pr_warn("failed to get cpu0 device\n");
-		return;
-	}
-	np = of_node_get(cpu_dev->of_node);
-	if (!np) {
-		pr_warn("failed to find cpu0 node\n");
-		return;
-	}
-
-	if (dev_pm_opp_of_add_table(cpu_dev)) {
-		pr_warn("failed to init OPP table\n");
-		goto put_node;
-	}
-
-	imx6ul_opp_check_speed_grading(cpu_dev);
-
-put_node:
-	of_node_put(np);
-}
-
 static inline void imx6ul_enet_init(void)
 {
 	imx6ul_enet_clk_init();
@@ -192,10 +101,8 @@ static void __init imx6ul_init_irq(void)
 
 static void __init imx6ul_init_late(void)
 {
-	if (IS_ENABLED(CONFIG_ARM_IMX6Q_CPUFREQ)) {
-		imx6ul_opp_init();
+	if (IS_ENABLED(CONFIG_ARM_IMX6Q_CPUFREQ))
 		platform_device_register_simple("imx6q-cpufreq", -1, NULL, 0);
-	}
 
 	imx6ul_cpuidle_init();
 }
-- 
2.17.1

